pageserver/src/virtual_file/owned_buffers_io/write/flush.rs

```diff
@@ -251,10 +261,47 @@ where
             }
 
             // Write slice to disk at `offset`.
-            let slice = self
-                .writer
-                .write_all_at(request.slice, request.offset, &self.ctx)
-                .await?;
+            //
+            // Error handling happens according to the current policy of crashing
+            // on fatal IO errors and retrying in place otherwise (deeming all other errors retryable).
+            // (The upper layers of the Pageserver write path are not equipped to retry write errors
+            //  becasuse they often deallocate the buffers that were already written).
+            //
+            // TODO: cancellation sensitiity.
+            // Without it, if we hit a bug where retrying is never successful,
+            // then we can't shut down the timeline/tenant/pageserver cleanrly because
+            // upp layers of the Pageserver write path are holding the gate open for EphemeralFile.
+            //
+            // TODO: use utils::backoff::retry once async closures are actually usable
```

Comment:
didn't async closures just get stabilized in 1.85?  (if this comment is calling out a specific outstanding issue then let's mention or link it)

Reply:
I tried in this PR, they're not good yet: https://github.com/neondatabase/neon/pull/10993/commits/cbf735426d5fcb9782a650a2e53cb0d3101920c3

https://neondb.slack.com/archives/C0277TKAJCA/p1740579692226909

Reply:
This is the commit where I gave up: https://github.com/neondatabase/neon/pull/10993/commits/a03f335121480afc0171b0f34606bdf929e962c5